core/actionresult/src/main/kotlin/nl/q42/template/actionresult/data/ActionResultMapper.kt

```diff
@@ -3,82 +3,98 @@ package nl.q42.template.actionresult.data
+import nl.q42.template.actionresult.domain.ActionResult
 
 private const val UNAUTHENTICATED_STATUSES = 401
 private const val NOT_FOUND_REQUESTS_STATUS = 404
 private const val TOO_MANY_REQUESTS_STATUS = 429
 
 private fun isUnAuthorized(httpStatusCode: Int?) = httpStatusCode == UNAUTHENTICATED_STATUSES
+
 private fun isNotFound(httpStatusCode: Int?) = httpStatusCode == NOT_FOUND_REQUESTS_STATUS
+
 private fun isTooManyRequests(httpStatusCode: Int?) = httpStatusCode == TOO_MANY_REQUESTS_STATUS
 
 /**
  * Maps a remote [NetworkResponse] to an [ActionResult] transforming all possible errors and exceptions into our domain error models
  */
-suspend fun <T : Any> mapToActionResult(request: suspend () -> NetworkResponse<T, ApiErrorResponse>): ActionResult<T> =
-    try {
-        request().networkResponseToActionResult()
-    } catch (e: Exception) {
-        if (e is CancellationException) ActionResult.Error.Cancelled(e) else ActionResult.Error.Other(e)
+suspend fun <T : Any> mapToActionResult(request: suspend () -> NetworkResponse<T, ApiErrorResponse>): ActionResult<T> = try {
+    request().networkResponseToActionResult()
+} catch (e: Exception) {
+    if (e is CancellationException) {
+        ActionResult.Error.Cancelled(
+            e
+        )
+    } else {
+        ActionResult.Error.Other(e)
     }
+}
 
-private fun <T : Any> NetworkResponse<T, ApiErrorResponse>.networkResponseToActionResult(): ActionResult<T> =
-    when (this) {
-        is NetworkResponse.Success -> ActionResult.Success(this.body)
-        is NetworkResponse.ServerError -> {
-            val errorMessage: String? = this.body?.message
-            val exception = Exception("$errorMessage $this.code")
-            when {
-                isNotFound(this.code) -> ActionResult.Error.NotFoundError
-                isTooManyRequests(this.code) -> ActionResult.Error.TooManyRequests(exception)
-                isUnAuthorized(this.code) -> ActionResult.Error.UnAuthorized(exception, errorMessage)
-                // a 404 is not a ServerError, but an UnknownError (mapping to Error.NotFoundError)
-                !errorMessage.isNullOrBlank() -> ActionResult.Error.ServerError(exception, errorMessage)
-                errorMessage == null -> ActionResult.Error.InvalidErrorResponse(exception, this.code)
-                else -> ActionResult.Error.Other(exception)
-            }
+private fun <T : Any> NetworkResponse<T, ApiErrorResponse>.networkResponseToActionResult(): ActionResult<T> = when (this) {
+    is NetworkResponse.Success -> ActionResult.Success(this.body)
+    is NetworkResponse.ServerError -> {
+        val errorMessage: String? = this.body?.message
+        val exception = Exception("$errorMessage $this.code")
+        when {
+            isNotFound(this.code) -> ActionResult.Error.NotFoundError
+            isTooManyRequests(this.code) -> ActionResult.Error.TooManyRequests(exception)
+            isUnAuthorized(
+                this.code
```

Comment:
yeah, this is not great

Reply:
max_line_length = 140 in [.editorconfig](https://github.com/Q42/Template.Android/pull/50/files#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5b)

https://pinterest.github.io/ktlint/latest/rules/standard/#max-line-length